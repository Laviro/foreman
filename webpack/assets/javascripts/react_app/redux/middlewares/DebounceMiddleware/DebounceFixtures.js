import { DEFAULT_DEBOUNCE } from './DebounceConstants';
import { withDebounce } from './DebounceHelpers';

export const type = 'SOME_TYPE';
export const key = 'SOME_KEY';
export const payload = { test: true };
export const debounce = DEFAULT_DEBOUNCE;
export const debounceID = 121;
export const initialState = {};
export const stateWithKey = { [key]: debounceID };
export const action = { type, key, payload };
export const actionWithDebounce = withDebounce(action);
