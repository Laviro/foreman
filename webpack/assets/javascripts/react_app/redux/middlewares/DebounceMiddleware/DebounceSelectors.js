export const selectDebounce = state => state.debounce || {};

export const selectDebounceItem = (state, key) => selectDebounce(state)[key];
