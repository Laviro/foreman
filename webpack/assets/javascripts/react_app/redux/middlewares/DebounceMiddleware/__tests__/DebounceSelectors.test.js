import { testSelectorsSnapshotWithFixtures } from '@theforeman/test';
import { key, stateWithKey } from '../DebounceFixtures';
import { selectDebounce, selectDebounceItem } from '../DebounceSelectors';

const state = {
  debounce: stateWithKey,
};

const fixtures = {
  'should return the debounce wrapper': () => selectDebounce(state),
  'should return the debounce item': () => selectDebounceItem(state, key),
};

describe('intervals selectors', () =>
  testSelectorsSnapshotWithFixtures(fixtures));
