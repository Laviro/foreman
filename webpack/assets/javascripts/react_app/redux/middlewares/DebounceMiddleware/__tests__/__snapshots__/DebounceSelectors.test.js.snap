// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`intervals selectors should return the debounce item 1`] = `121`;

exports[`intervals selectors should return the debounce wrapper 1`] = `
Object {
  "SOME_KEY": 121,
}
`;
